contrib/platform/src/com/sun/jna/platform/win32/Netapi32.java

```diff
@@ -42,6 +44,82 @@
 
     Netapi32 INSTANCE = Native.load("Netapi32", Netapi32.class, W32APIOptions.DEFAULT_OPTIONS);
 
+    int MAX_PREFERRED_LENGTH = -1;
+
+    /**
+     * Contains information about the session, including name of the computer; name
+     * of the user; and active and idle times for the session.
+     */
+    @FieldOrder({ "sesi10_cname", "sesi10_username", "sesi10_time", "sesi10_idle_time" })
+    class SESSION_INFO_10 extends Structure {
+        public String sesi10_cname;
+        public String sesi10_username;
+        public int sesi10_time;
+        public int sesi10_idle_time;
+
+        public SESSION_INFO_10() {
+            super(W32APITypeMapper.DEFAULT);
+        }
+
+        public SESSION_INFO_10(Pointer p) {
+            super(p, Structure.ALIGN_DEFAULT, W32APITypeMapper.DEFAULT);
```

Comment:
The DEFAULT Type mapper correct here? From the header file:

> //
// Only the UNICODE version of the LM APIs are available on NT.
// Non-UNICODE version on other platforms
//
#if defined( _WIN32_WINNT ) || defined( WINNT ) || defined( __midl ) \
    || defined( FORCE_UNICODE )
#define LMSTR   LPWSTR
#define LMCSTR  LPCWSTR
#else
#define LMSTR   LPSTR
#define LMCSTR  LPCSTR
#endif

I risk a guess, on recent Windows Versions (i.e. everything post W2K) - is WIN32_WINNT or WINNT defined? If so, this would need the UNICODE mapper, if not some more logic is necessary.

Reply:
Good catch. Those strings are always Unicode.

Reply:
So if they are mapped directly to `WString`, we don't need any type mapper.